### PR TITLE
use requestAnimationFrame

### PR DESCRIPTION
**Pull request by greggman**
This is my attempt to refactor processing to use requestAnimationFrame instead of setInterval.

See http://paulirish.com/2011/requestanimationframe-for-smart-animating/

### CODE REVIEW

File: processing.js [hunk 1]
```diff
@@ -35,6 +35,18 @@
...
 
+  /* handle animation without DOSing the user's browser */
+  var requestAnimFrame = (function() {
+    return  window.requestAnimationFrame       || 
+            window.webkitRequestAnimationFrame || 
+            window.mozRequestAnimationFrame    || 
+            window.oRequestAnimationFrame      || 
+            window.msRequestAnimationFrame     || 
+            function(/* function */ callback, /* DOMElement */ element){
+              window.setTimeout(callback, 1000 / 60);
```
Comment (jbuck) [inline processing.js:46]: Change 1000 / 60 to curMsPerFrame

**Comment by jbuck** [comment]
Hi greggman, thanks for submitting this patch. This patch looks good to me, we just need to figure out why RAF goes nuts and attempts to draw at 150 FPS when changing the frame rate with a frameRate(int desiredFPS) call. Any ideas?

You can see this with the example in /processing-js/examples/seneca/bug1116/ just have the frameRate change on keyPressed or something similar.

**Comment by jbuck** [comment]
As an alternative, we could just revert to the old setInterval behaviour if the user calls frameRate().

**Comment by greggman** [comment]
I hope I did the right thing as far as git is concerned.

I believe this fixes the issue.

**Comment by greggman** [comment]
Doh! I picked "Comment & Close" instead of just "Comment" :-(

**Comment by greggman** [comment]
Sorry, I took that comment about using curMsPerFrame without actually checking it.

In order to do that I'd have to move requestAnimFrame somewhere inside the Processing object. Either that or add frameRate argument to requestAnimFrame. Can do either or leave it.

**Comment by arxpoetica** [comment]
Why is this still open, when this is so important?

http://my.opera.com/emoller/blog/2011/12/20/requestanimationframe-for-smart-er-animating

**Comment by arxpoetica** [comment]
Putting my money where my mouth is. I started w/ this patch and tried to add cancelAnimationFrame to the mix. Buggy, but I'll add a link to the patch/work soon.

**Comment by arxpoetica** [comment]
I'm confused. Why was this request closed? It's still a very valid request.

**Comment by lennerd** [comment]
+1 for having better performance over all.

**Comment by jwdunn1** [comment]
+1 for this important update (the latest v1.4.7 is still using setInterval)

**Comment by leoxi** [comment]
+1

**Comment by arxpoetica** [comment]
Still no answer on why this was closed? Disappointing.

**Pull request closed without merging**